gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java

```diff
@@ -1030,14 +1035,22 @@ private StorageResourceId getPrefixId(URI prefix) {
       }
     }
 
+    StorageResourceId dirId =
+        StorageResourceId.fromUriPath(UriPaths.toDirectory(path), /* allowEmptyObjectName= */ true);
+
     List<GoogleCloudStorageItemInfo> dirItemInfos =
         dirId.isRoot()
             ? gcs.listBucketInfo()
-            : gcs.listObjectInfo(dirId.getBucketName(), dirId.getObjectName());
-    if (dirItemInfos.isEmpty() && !gcs.getItemInfo(dirId).exists()) {
+            : gcs.listObjectInfo(
+                dirId.getBucketName(), dirId.getObjectName(), LIST_FILE_INFO_LIST_OPTIONS);
+    if (pathId.isStorageObject() && dirItemInfos.isEmpty()) {
       throw new FileNotFoundException("Item not found: " + path);
     }
 
+    if (!dirItemInfos.isEmpty() && Objects.equals(dirItemInfos.get(0).getResourceId(), dirId)) {
```

Comment:
It checks whether prefix object was listed and removes it from the result of this is the case, because `listFileInfo` should list only objects in directory if they exist not directory object itself.

Reply:
Right I understand that, I was particularly asking if plain `!dirItemInfos.isEmpty()` would just work, I could be wrong but as I understand it the list always returns the prefix itself?

Reply:
I think it will not return prefix object info if prefix object does not exist and if implicit directory inference is not enabled.